src/libstore/binary-cache-store.cc

```diff
@@ -434,7 +434,9 @@ StorePath BinaryCacheStore::addTextToStore(const string & name, const string & s
     if (!repair && isValidPath(path))
         return path;
 
-    auto source = StringSource { s };
+    StringSink sink;
+    dumpString(s, sink);
+    auto source = StringSource { *sink.s };
```

Comment:
This looks like a bugfix we should PR sooner?

Reply:
We could yes. Though I don't think this method was used anywhere before this PR so probably not an emergency